modules/microshift-greenboot-dir-structure.adoc

```diff
@@ -0,0 +1,38 @@
+// Module included in the following assemblies:
+//
+// * microshift_running applications/microshift-greenboot.adoc
+
+:_content-type: CONCEPT
+[id="microshift-greenboot-dir-structure_{context}"]
+= How greenboot uses directories to run scripts
+
+Health check scripts run from four `/etc/greenboot` directories. These scripts run in alphabetical order. Keep this in mind when you configure the scripts for your workloads.
+
+When the system starts up, greenboot will run the scripts in the `required.d` and `wanted.d` directories. Depending on the outcome of those scripts, greenboot continues the start-up or attempts a rollback as follows:
+
+. System as expected: When scripts in the `required.d` directory are successful, greenboot runs any scripts present in the `/etc/greenboot/green.d` directory.
+
+. System trouble: If any of the scripts in the `required.d` directory fail, greenboot will run any pre-rollback scripts present in the `red.d` directory, then restart the system.
+
+[NOTE]
+====
+Greenboot redirects script and health check output to the system log. When you are logged in, a "message of the day" provides the overall system health output.
+====
+
+[id="greenboot-directories-details_{context}"]
+== Greenboot directories details
+
+Returning a non-zero sum for any script means that script has failed. Greenboot retries scripts before declaring a failure.
+
+* `/etc/greenboot/check/required.d` contains the health checks that must not fail. If the scripts fail, greenboot retries them three times by default. You can configure the number of retries.
+//Q: How do users configure the number of retries?
```

Comment:
Retries can be configured in /etc/greenboot/greenboot.conf by setting GREENBOOT_MAX_BOOTS=5 for example. The default number of reboots is 3.